app/javascript/fetchProc.js

```diff
@@ -411,12 +411,21 @@ export const downloadFile = (route, appState) =>{
   })
   .then(resp => {
     if(download){
-      let url = URL.createObjectURL(resp);
-      var a = document.createElement('a');
-      a.href = url;
-      a.download = filename;
-      a.click();
-      URL.revokeObjectURL(url);
+      if (window.navigator && window.navigator.msSaveOrOpenBlob) {
+        window.navigator.msSaveOrOpenBlob(resp, filename);
+      }
+      else{
+        let url = URL.createObjectURL(resp);
```

Comment:
wow, what's this? how about a one line comment saying at least what corner case this  is.